### ISSUE TITLE
Oxygen Vent Only fills 98 blocks

### ISSUE THREAD

**Issue by tprussak** [zmaster587/AdvancedRocketry#1485]
## Version of Advanced Rocketry
1.4.1-121
## Have you verified this is an issue in the latest unstable build
Y
## Version of LibVulpes
0.3.0-1
## Version of Minecraft
1.12.2

## Description of the problem
The oxygen vent only works when it is filling 98 blocks or less. When it needs to fill the 99th block, the entire area goes back to vacuum. This was tested with radii from 32 to 100, and all four calculation methods. 
![2019-01-18_16 33 33](https://user-images.githubusercontent.com/1276028/51418513-13c4ce80-1b41-11e9-81b3-fb8ba74d2587.png)
5 by 5 by 4 space, 3 blocks taken up by oxygen vent, ender tank of oxygen, and power. One more block is taken out from the side wall, reaching 98 bocks.
![2019-01-18_16 33 38](https://user-images.githubusercontent.com/1276028/51418566-761dcf00-1b41-11e9-8861-1cad6696bec9.png)
Removing one more block from the wall, now with 99 blocks of space, causes the room to be a vacuum. The oxygen vent still makes a hissing noise as if it were still working.

**Comment by mikewk** [comment]
I just ran into this in Enigmatic 2: Expert as well.  I have tracked the problem though, I believe.  The relevant section of code in TileOxygenVent.java.

```
if (amtToDrain > 0 && amtToDrain < 1)
    amtToDrain = 1;
FluidStack drainedFluid = this.drain((int)amtToDrain, false);

if( (drainedFluid != null && drainedFluid.amount >= amtToDrain) || amtToDrain == 0) {
```

If I'm reading this correctly and based off the recent commits/releases, the fix to allow small rooms to consume oxygen breaks large rooms.  The culprit is casting the amtToDrain to int in the drain call while not casting it to int in the comparison.  This causes drainedFluid to always be smaller than the uncasted amtToDrain unless it's exactly even, which is very unlikely if the room size is large but guaranteed if the room size is small due to the first 2 lines of the fix.

I don't think there's any simple fix that won't cause there to be discontinuities in the amount of oxygen consumed vs room size, since the Forge tank requires consumption to be an int.  I'd say the easiest fix is to just cast amtToDrain to an int in the comparison as well.  

For tprussak, in the short term, I'm using 2 CO2 scrubbers because it bring amtToDrain down to zero and therefore bypasses the bug.

I apologize if any of the details are wrong, it's been forever since I've done Java.  But I was bored and thought I could figure it out and I'm pretty sure the broad strokes are right.
